compiler/frontend/src/mir_optimize/cleanup.rs

```diff
@@ -62,7 +62,6 @@ impl Mir {
                 (Expression::Builtin(a), Expression::Builtin(b)) => {
                     format!("{a:?}").cmp(&format!("{b:?}"))
                 }
-                (Expression::Symbol(a), Expression::Symbol(b)) => a.cmp(b),
```

Comment:
Something like this should work: `a.symbol.cmp(b.symbol).then_with(|| a.value.cmp(b.value))`

Reply:
For now, tags with values are not constant-lifted and not reordered because they may reference other expressions and thereby get invalid by reordering, even if they're constant.
Only tags without values are moved. I fixed it so that they are also sorted according to the symbol.